core/src/main/scala/org/apache/spark/FutureAction.scala

```diff
@@ -105,6 +108,7 @@ trait FutureAction[T] extends Future[T] {
  * A [[FutureAction]] holding the result of an action that triggers a single job. Examples include
  * count, collect, reduce.
  */
+@DeveloperApi
```

Comment:
Wait a second... I don't think we should do this; AFAIK we intend for this to be a stable API and this is going to bypass the MiMa checks. If MiMa broke due to changes in this patch then the right fix is to add specific MiMa excludes for those changes, not to make the whole thing into a `DeveloperAPI`.